octopusdeploy_framework/framework_provider.go

```diff
@@ -72,9 +72,29 @@ func (p *octopusDeployFrameworkProvider) DataSources(ctx context.Context) []func
 }
 
 func (p *octopusDeployFrameworkProvider) Resources(ctx context.Context) []func() resource.Resource {
-	return []func() resource.Resource{
+	resources := []func() resource.Resource{
 		NewProjectGroupResource,
 	}
+
+	resources = append(resources, getImportableResources()...)
+
+	return resources
+}
+
+func getImportableResources() []func() resource.Resource {
+	importableResources := []func() resource.ResourceWithImportState{
+		NewLifecycleResource,
+	}
+
+	resources := make([]func() resource.Resource, 0, len(importableResources))
+
+	for _, r := range importableResources {
+		resourceFunc := r
+		resources = append(resources, func() resource.Resource {
+			return resourceFunc()
+		})
+	}
+	return resources
```

Comment:
When the `ResourceWithImportState` is removed none of this is required, you can just the `NewLifecycleResource` func reference here in the collection in `Resources`